### PR TITLE
Improve agent navigation and command output

### PR DESCRIPTION
**Pull request by neekolas**
## Summary

- Replace regex-based `just outline` with locked ast-outline 1.9.0 and upstream defaults. Document optional `--no-docs --no-fields --no-attrs` flags without adding a compact mode. Add `just show` for symbol reads.
- Pin RTK 0.49.0 in the default and Rust Nix shells. Apply dedicated Cargo filters inside selected recipes, with a raw-output bypass.
- Enable compact output from the repo agent hook unless explicitly disabled. Keep CI, custom coverage commands, structured output, and unsupported commands raw. Nextest filtering needs a separate opt-in.
- Document navigation, filter limits, and recovery. Add tests for argument boundaries, exit status, bypass behavior, and language coverage.

Approved plan: https://plan.ref.tools/g7vVmTVfh6TfvsHa

- Audit all 40 tracked `AGENTS.md` files. Keep shared workflow rules in parent files, remove repeated guidance, and route focused command examples through the existing recipes or optional runner.

## Validation

- Passed: `just agent-test` (27 tests, including default output and optional compact flags).
- Passed: `just lint-config lint-markdown lint-proto`, `cargo fmt --check`, and both Hakari checks.
- Passed: 148 documented Just command dry runs. No long exact paragraphs remain duplicated between parent and child agent files.
- Passed: ShellCheck for both new launchers and `git diff --check`.
- Built RTK 0.49.0 with Nix on aarch64-darwin and verified the shell selects that package.
- Evaluated Rust shells on x86_64-linux and aarch64-linux, and the default shell on x86_64-linux. Linux builds were not run locally.
- Passed: real source outline and `just check crate xmtp_macro` with filtering enabled and disabled.
- After rebasing onto `self-hosted` at `9b9af7c58`, full `just lint` passes. The upstream Apple toolchain fix resolves the earlier SDK linker error. Both C and Rust checks in `dev/check-apple-toolchain` pass, and all 26 helper tests pass again.

## Limits

Outlines are syntax-based and can omit declarations. ast-outline returns zero for some user errors, so agents must inspect its notes. RTK Nextest filtering can hide short early errors; it remains off by default. Filtered failures keep their status and print a raw-output recovery instruction. Commands are never retried automatically.

No CI workflow, Rust toolchain, or public SDK API changes.

> [!NOTE]
> ### Add `dev/agent-run` command wrapper and `ast-outline` navigation
> - Add `dev/agent-run` wrapper to filter supported Cargo commands through RTK for compact output. It bypasses filtering for CI, inspection modes, or missing RTK.
> - Replace ripgrep-based `just outline` with `dev/ast-outline` launcher and add `just show` recipe for code navigation. This supports multiple paths and preserves arguments.
> - Update [justfile](https://github.com/xmtp/libxmtp/pull/4144/files#diff-deb9bb56fb122db0b605aa5b63f95a4665c905b18dd670e1fa6c877576a94ff1) and [backend.just](https://github.com/xmtp/libxmtp/pull/4144/files#diff-a8e160e8b7d6b12d00ac75bbaecc638d6095068e7b81b8b093bf73f72ed09f28) to route `cargo test`, `cargo check`, and `lint-rust` through `dev/agent-run`.
> - Add RTK 0.49.0 Nix derivation and include it in the `agentTools` package group for local and Rust shells.
> - Risk: `just outline` delegates to `ast-outline` instead of ripgrep, changing its default output format to include line ranges and omit function bodies. In-worktree Bash commands default to `XMTP_RTK=1` when unset.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized a7e94f5.</sup>
> 
>

### CODE REVIEW

File: apps/backend/backend.just [hunk 1]
```diff
@@ -32,7 +32,7 @@ run:
 # Backend unit, RPC, storage, replica, and ingress tests. Needs `db-up`.
 test *args="":
     {{ _env }} && \
-      RUST_TEST_THREADS="${RUST_TEST_THREADS:-4}" cargo test --locked -p xmtp_backend {{ args }}
+      RUST_TEST_THREADS="${RUST_TEST_THREADS:-4}" {{ root }}/dev/agent-run cargo test --locked -p xmtp_backend {{ args }}
```
[the remark this answers is not in the document]
Comment (neekolas) [reply apps/backend/backend.just]: Fixed in a7e94f531: quoted the launcher path. Verified the rendered command with a root path containing spaces. Full just lint passes.
